### PR TITLE
Remove incubator content, touchup existing requirements

### PR DESCRIPTION
**Pull request by spiffxp**
Trying to bring this repo in alignment with the way things are today:
- we keep talking about how incubator is / should going away, so... drop wording about it?
- very few repos in kubernetes/ have RELEASE.md files
- almost every repo now uses the exact code-of-conduct.md from this repo
- [docs repos don't use the apache 2 license](https://github.com/kubernetes/community/blob/master/governance.md#repository-guidelines)
- OWNERS files point to owners docs
- remove reference to godep from CONTRIBUTING.md (not all repos are golang, not all golang repos have chosen to use that)
- link to to canonical CLA instructions at https://git.k8s.io/community/CLA.md

This is in the interest of making this repo a living example of a "correct" repo, per https://github.com/kubernetes/community/pull/1569

### CODE REVIEW

File: README.md
Comment (cblecker) [inline README.md]: needs `https://` in link

File: README.md
Comment (mattfarina) [inline README.md]: The document notes it's for things on the github.com/kubernetes organization. But, it's being proposed for uses outside of that (other orgs). Should that be updated?

Comment (spiffxp) [reply README.md]: I would like to change the wording to be something like "new projects in a kubernetes github organization" with a link to a list of those organizations.  But we don't yet have that list merged.  I was planning to do so in a follow on PR.

**Comment by cblecker** [comment]
/lgtm
/hold

Looks good to me. Hold if you wanted other feedback before merge, @spiffxp.

**Comment by spiffxp** [comment]
/cc @mattfarina @michelleN 
PTAL

@kubernetes/steering-committee if I hear no objections I will assume you're OK with this as well

**Review by mattfarina — APPROVED** [review]

**Comment by bgrant0607** [comment]
LGTM

**Pull request merged**
